apps/root/src/pages/transfer/components/recipient-address/components/add-contact-modal/index.tsx

```diff
@@ -0,0 +1,227 @@
+import useStoredContactList from '@hooks/useStoredContactList';
+import useValidateAddress, { ValidationOutput } from '@hooks/useValidateAddress';
+import CenteredLoadingIndicator from '@common/components/centered-loading-indicator';
+import styled from 'styled-components';
+import ErrorCircleIcon from 'ui-library/src/icons/errorCircle';
+
+interface AddContactModalProps {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  goBackToTransfer: () => void;
+}
+
+const StyledStatusTitle = styled(Typography).attrs({ variant: 'h5' })`
+  ${({ theme: { palette } }) => `
+  font-weight: bold;
+  color: ${colors[palette.mode].typography.typo1};
+  text-align: center;
+`}
+`;
+
+const StyledStatusDescription = styled(Typography).attrs({ variant: 'body1' })`
+  ${({ theme: { palette } }) => `
+  color: ${colors[palette.mode].typography.typo2}
+  text-align: center;
+`}
+`;
+
+const inputRegex = RegExp(/^[A-Fa-f0-9x]*$/);
+
+const PostContactStatusContent = ({
+  icon,
+  title,
+  description,
+  button,
+}: {
+  icon: React.ReactElement;
+  title: React.ReactElement;
+  description: React.ReactElement;
+  button: React.ReactElement;
+}) => (
+  <Grid container direction="column" rowGap={6}>
+    <Grid item xs={12}>
+      {icon}
+    </Grid>
+    <Grid container rowGap={2} direction="column">
+      <StyledStatusTitle>{title}</StyledStatusTitle>
+      <StyledStatusDescription>{description}</StyledStatusDescription>
+    </Grid>
+    <Grid item xs={12}>
+      {button}
+    </Grid>
+  </Grid>
+);
+
+const AddContactModal = ({ open, setOpen, goBackToTransfer }: AddContactModalProps) => {
+  const intl = useIntl();
+  const contactListService = useContactListService();
+  const contactList = useStoredContactList();
+  const [contactAddress, setContactAddress] = React.useState<string>('');
+  const [contactLabel, setContactLabel] = React.useState<string>('');
+  const [postContactStatus, setPostContactStatus] = React.useState<'loading' | 'success' | 'error' | null>(null);
+
+  React.useEffect(() => {
+    if (!open) {
+      setPostContactStatus(null);
+      setContactAddress('');
+      setContactLabel('');
+    }
+  }, [open]);
+
+  const validateContactRepetition = React.useCallback<() => ValidationOutput>(() => {
+    if (contactList.some((contact) => contact.address === contactAddress.toLowerCase())) {
+      return {
+        errorMessage: intl.formatMessage(
+          defineMessage({
+            defaultMessage: 'Contact already exists',
+            description: 'contactAlreadyExists',
+          })
+        ),
+        isValidAddress: false,
+      };
+    }
+    return {
+      isValidAddress: true,
+      errorMessage: '',
+    };
+  }, [contactList, contactAddress]);
+
+  const { isValidAddress, errorMessage } = useValidateAddress({
+    address: contactAddress,
+    additionalValidations: [validateContactRepetition],
+  });
+
+  const onChangeAddress = (nextValue: string) => {
+    if (!inputRegex.test(nextValue)) {
```

Comment:
if we have the useValidateAddress, do we need to parse here?

Reply:
Mmm I think I can make the input controlled by this hook! I'll try it out